kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/Worker.java

```diff
@@ -112,6 +112,7 @@ protected boolean receive(Envelope envelope) {
         writers.values().stream().flatMap(writer -> writer.complete().stream()).collect(toList());
     Map<TopicPartition, Offset> offsets = Maps.newHashMap(sourceOffsets);
 
+    writers.values().forEach(RecordWriter::close);
```

Comment:
`writer.complete()` on 112 does call `close()`.   Why the second check?

Reply:
Can you show me this offline? 
Intellij is showing me `RecordWriter::close` is only called in 2 places, neither of which cover this code path AFAICT, so I'm confused. 
<img width="483" alt="Screenshot 2024-03-22 at 10 52 19 AM" src="https://github.com/tabular-io/iceberg-kafka-connect/assets/160629711/d3df642a-2f17-4769-93a1-5e707ed09e14">

Reply:
`flush` calls `writer.complete()`, `writer.complete()` itself calls `close` on itself.

Reply:
We still have a memory leak here though. 

If a writer fails to close, exception thrown which means later writers don't get closed.  We should put something in that calls close() and collects all errors, but ensures that `close` is called on every writer.  Otherwise we just clear the writer but I'm not convinced the writer object itself gets GC'd away until close it called --there might be some state stuck around in the JVM that prohibits it from being GC'd.

ValidatedNonEmptyList :(